providers/flagd/src/main/java/dev/openfeature/contrib/providers/flagd/resolver/process/storage/connector/sync/SyncStreamConnector.java

```diff
@@ -161,4 +153,21 @@ void observeEventStream(final BlockingQueue<QueuePayload> writeTo, final AtomicB
 
         log.info("Shutdown invoked, exiting event stream listener");
     }
+
+    private void restart() {
+        log.debug("starting our new restart method");
+        FlagSyncServiceStub stub =
+                FlagSyncServiceGrpc.newStub(this.grpcConnector.getChannel()).withWaitForReady();
+        if (streamDeadline > 0) {
+            stub = stub.withDeadlineAfter(streamDeadline, TimeUnit.MILLISECONDS);
+        }
```

Comment:
as those settings are the same, we might want to extract this stub creation into a util function or something similar. We will most likely improve something for one GRPC Stream connection, but forget about the other one, afterwards. Having this unified might be a good idea.

Reply:
These methods are a bit too different IMO to commonize, they call different stream methods etc. I've moved the stub creation to the constructor though, there's no reason to create one every time.